spec/acceptance/class_spec.rb

```diff
@@ -1,16 +1,61 @@
 require 'spec_helper_acceptance'
 
-describe 'nodejs class:', unless: UNSUPPORTED_PLATFORMS.include?(fact('osfamily')) do
-  it 'runs successfully' do
-    pp = <<-EOS
-    class { 'nodejs': }
-    if $::osfamily == 'RedHat' and $::operatingsystemrelease =~ /^5\.(\d+)/ {
-      class { 'epel': }
-      Class['epel'] -> Class['nodejs']
-    }
-    EOS
+describe 'nodejs class:', unless: UNSUPPORTED_PLATFORMS.include?(fact('os.family')) do
+  case fact('os.family')
+  when 'RedHat'
+    pkg_cmd = 'yum info nodejs | grep "^From repo"'
+  when 'Debian'
+    pkg_cmd = 'dpkg -s nodejs | grep ^Maintainer'
+  end
+
+  context 'default parameters' do
+    it 'runs successfully' do
+      pp = <<-EOS
+      class { 'nodejs': }
+      if $::osfamily == 'RedHat' and $::operatingsystemrelease =~ /^5\.(\d+)/ {
+        class { 'epel': }
+        Class['epel'] -> Class['nodejs']
+      }
+      EOS
+      # Run it twice and test for idempotency
```

Comment:
I like https://github.com/theforeman/foreman-installer-modulesync/blob/0ee390953528a18b48ffdf94a97b71aa5029b1ad/moduleroot/spec/spec_helper_acceptance.rb.erb#L39-L47

Reply:
I like this too, but can't use shared examples in an 'it' block, and maybe overkill to use within the same file? I am going to switch to the catch_failures / catch_changes pattern and simplify things a bit.

Reply:
That does work: https://github.com/theforeman/puppet-foreman/blob/f0cfa1f39de13c1e937d23f2638db858eb7d2d27/spec/acceptance/foreman_cli_spec.rb#L33